stack.yaml

```diff
@@ -1,98 +1,12 @@
-resolver: ghc-8.4.1
-compiler: ghc-8.4.1
-compiler-check: match-exact
+resolver: nightly-2018-04-01
 
 extra-deps:
-- Glob-0.9.2
-- doctest-0.14.1
-- gauge-0.2.1
-- hedgehog-0.5.2
-- markdown-unlit-0.5.0
-- tasty-0.12.0.1
 - tasty-hedgehog-0.1.0.2
-- tasty-hspec-1.1.3.3
 - tiempo-0.0.1.1
 - time-units-1.0.0
-- type-spec-0.3.0.1
 
-## needed for above libraries
-- async-2.1.1.1
-- base-4.10.1.0
-- base-compat-0.9.3
-- basement-0.0.7
-- clock-0.7.2
-- code-page-0.1.3
-- concurrent-output-1.10.4
-- dlist-0.8.0.4
-- exceptions-0.8.3
-- ghc-paths-0.1.0.9
-- hspec-2.4.8
-- hspec-core-2.4.8
-- lifted-async-0.9.3.3
-- mmorph-1.1.1
-- monad-control-1.0.2.3
-- optparse-applicative-0.14.2.0
-- pretty-show-1.6.16
-- primitive-0.6.3.0
-- random-1.1
-- resourcet-1.2.0
-- semigroups-0.18.4
-- syb-0.7
-- tagged-0.8.5
-- tasty-quickcheck-0.9.2
-- tasty-smallcheck-0.8.1
-- th-lift-0.7.8
-- time-1.5.0.1
-- transformers-base-0.4.4
-- transformers-compat-0.6.0.6
-- unbounded-delays-0.1.1.0
-- vector-0.12.0.1
-- wl-pprint-annotated-0.1.0.0
-
-## needed for above libraries
-- HUnit-1.6.0.0
-- QuickCheck-2.11.3
-- ansi-terminal-0.8.0.2
-- ansi-wl-pprint-0.6.8.2
-- call-stack-0.1.0
-- constraints-0.10
-- haskell-lexer-1.0.1
-- hspec-discover-2.4.8
-- hspec-expectations-0.8.2
-- lifted-base-0.2.3.11
-- quickcheck-io-0.2.0
-- regex-base-0.93.2
-- setenv-0.1.1.3
-- smallcheck-1.1.3.1
-- terminal-size-0.3.2.1
-- tf-random-0.5
-- unliftio-core-0.1.1.0
-
-# needed for above
-- colour-2.3.4
-- hashable-1.2.7.0
-- logict-0.6.0.2
-
-- git: https://github.com/hvr/regex-tdfa
-  commit: b7189c6972b7c7a000eece503ae632cbc273f003
-
-# flags
-- aeson-1.3.0.0
+## Serialise
 - serialise-0.2.0.0
-
-# for flags
-- attoparsec-0.13.2.2
 - cborg-0.2.0.0
-- half-0.2.2.3
-- scientific-0.3.5.2
-- th-abstraction-0.2.6.0
-- time-locale-compat-0.1.1.3
-- unordered-containers-0.2.9.0
-- uuid-types-1.0.3
-
-- integer-logarithms-1.0.2.1
-- old-locale-1.0.0.7
-
-- happy-1.19.9
 
 allow-newer: true
```

Comment:
Is `allow-newer` still needed? Also, do we need to patch `.travis.yml` or not? I don't remember...

I guess, CHANGELOG update is not required.. And after this PR is merged we can safely upload `1.0.0` version to Hackage!

Reply:
Well, I didn't try to remove it, but then `base` and a lot of other packages should be added in `stack.yaml`.. Should I try?